ui/admin/tests/acceptance/host-catalogs/hosts/read-test.js

```diff
@@ -0,0 +1,94 @@
+import { module, test } from 'qunit';
+import { visit, currentURL, find } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+import a11yAudit from 'ember-a11y-testing/test-support/audit';
+import {
+  authenticateSession,
+  // These are left here intentionally for future reference.
+  //currentSession,
+  //invalidateSession,
+} from 'ember-simple-auth/test-support';
+
+module('Acceptance | host-catalogs | hosts | read', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  const instances = {
+    scopes: {
+      global: null,
+      org: null,
+      project: null,
+      hostCatalog: null,
+      host: null,
+    },
+  };
+  const urls = {
+    globalScope: null,
+    orgScope: null,
+    projectScope: null,
+    hostCatalogs: null,
+    hostCatalog: null,
+    hosts: null,
+    host: null,
+    unknownHost: null,
+    newHost: null,
+  };
+
+  hooks.beforeEach(function () {
```

Comment:
@DhariniJeeva @randallmorey @gsusmi 
I am writing this as a question, not than as a statement: I see a recurrent pattern on all the capabilities tests. We are creating a test file per capability, which is a nice separation of concerns. But we are also repeating code through these files, all the instances, urls and setting those within the beforeEach statements are the same across capability.
I wonder if we could create a test helper file per capability where we define and export the repeating patterns, so then the test file just import those and we avoid repeating code.

For this use case, could be something like read-capability-test-helper.js? exports instances and urls initialization (basically generating the object equal to null) and generate all the resources.

Let me know your thoughts, perhaps this is not a good idea.

Reply:
This is something I've thought about as well.  Essentially we need a standard way to model a test scenario and all the resources that go into it.  And from there, we might be able to move the standard test cases themselves into helpers.  This is housekeeping and hopefully we can find time in the future.

Reply:
Sure! yeah I should mentioned before this message was OUT of topic of this PR and more a housekeeping thing.
Thanks!

Reply:
Thanks for bringing this up @calcaide.